### ISSUE TITLE
question about rewriting loss function or data input in GAN model

### ISSUE THREAD

**Issue by naiveyueya** [LouisChen1992/Deep_Learning_in_Asset_Pricing#3]
hello, i have a question about lossfunction in GAN model. In your paper, the outside of loss function is permno number N, not time T. This means the input must be N*(T*Firm_characteristics_number), and batchsize number is N. That is, you train model in one stock of time t1-tn, and then retrain it in next stock. Or not, can you give me a tips to rewrite the loss function? 
Thank you for your sharing!

**Comment by LouisChen1992** [comment]
> hello, i have a question about lossfunction in GAN model. In your paper, the outside of loss function is permno number N, not time T. This means the input must be N*(T*Firm_characteristics_number), and batchsize number is N. That is, you train model in one stock of time t1-tn, and then retrain it in next stock. Or not, can you give me a tips to rewrite the loss function?
> Thank you for your sharing!

The input is a tensor of shape N * T * K. N is the number of all available stocks. There is no batch size here.

**Comment by naiveyueya** [comment]
Thank you a lot for your kind reply! 
But there is another question. SDF net needs an N_t * K input of time t to calculate  SDF M_{t+1}. The shape of SDF net input and Conventional input is different. So the input shape between SDF net and Convention net may be different. But in your paper,  inputs of two nets are the same. This confused me a lot. 
What's more, some stocks have discrete return list, that means when calculating loss funtion, each stock may need an index to show when it have R. Do I get it right?
Thank you!!

**Comment by LouisChen1992** [comment]
> Thank you a lot for your kind reply!
> But there is another question. SDF net needs an N_t * K input of time t to calculate SDF M_{t+1}. The shape of SDF net input and Conventional input is different. So the input shape between SDF net and Convention net may be different. But in your paper, inputs of two nets are the same. This confused me a lot.
> What's more, some stocks have discrete return list, that means when calculating loss funtion, each stock may need an index to show when it have R. Do I get it right?
> Thank you!!

The input for SDF network and Conditional network is the same. It is true that calculating SDF at time t requires N_t * K input. The conditional variable g_i for stock i requires 1 * K input and we consider those moment conditions for all N_t stocks at the same time. 

Yes you a mask of shape T * N to show whether a stock return exists at time t. It has value 1 if it exists and 0 otherwise.

**Comment by naiveyueya** [comment]
Thank you a lot!! I have no other questions now.
